src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good that you're using `useNavigation` from `@react-navigation/core`. This is a better practice than passing navigation as a prop because it makes the component less dependent on its parent and more reusable.



<!-- This is an auto-generated comment by OSS CodeRabbit -->